net/portmapper/igd_test.go

```diff
@@ -88,6 +99,9 @@ func (d *TestIGD) serveUPnPDiscovery() {
 	buf := make([]byte, 1500)
 	for {
 		n, src, err := d.upnpConn.ReadFrom(buf)
+		if !d.doUPnP {
```

Comment:
you should move this later so you still increment the counter but don't just don't reply.

then we can test whether clients are behaving (e.g. not sending when we don't want them to send).